lib/views/group_screen.dart

```diff
@@ -238,77 +262,116 @@ class _GroupScreenState extends State<GroupScreen>
                                               return Container(
                                                   alignment: Alignment.center,
                                                   child: posts.length == 0
-                                                      ? SingleChildScrollView(
-                                                          physics:
-                                                              AlwaysScrollableScrollPhysics(),
-                                                          child: Column(
-                                                            children: [
-                                                              Text(
-                                                                'You haven\'t joined or created any beacon yet',
-                                                                textAlign:
-                                                                    TextAlign
-                                                                        .center,
-                                                                style: TextStyle(
-                                                                    color:
-                                                                        kBlack,
-                                                                    fontSize:
-                                                                        20),
-                                                              ),
-                                                              SizedBox(
-                                                                height: 2.h,
-                                                              ),
-                                                              RichText(
-                                                                text: TextSpan(
-                                                                  // textAlign:
-                                                                  //   TextAlign
-                                                                  //       .center,
+                                                      ? CustomRefreshIndicator(
```

Comment:
Instead of adding multiple refresh indicators for tabs with both 0 or more than 0 posts, couldn't you add it simply to the tab and then check the condition of 0 posts?